bin/check-apache-status.rb

```diff
@@ -0,0 +1,120 @@
+#! /usr/bin/env ruby
+#
+#   check-apache-status
+#
+# DESCRIPTION:
+#   Test whether apache is up and functional by using mod_status
+#
+# OUTPUT:
+#   plain text
+#
+# PLATFORMS:
+#   Linux
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#
+# USAGE:
+# check-apache-status.rb --help
+# NOTES:
+#
+# LICENSE:
+# Magic Online http://www.magic.fr  @2016
+# Released under the same terms as Sensu (the MIT license); see LICENSE  for details.
+#
+
+require 'sensu-plugin/check/cli'
+require 'open3'
+require 'net/http'
+require 'net/https'
+
+#
+# Check  Apache Status
+#
+class CheckApacheStatus < Sensu::Plugin::Check::CLI
+  option :busy, long: '--busy BUSYWORKERS', short: '-b BusyWorkers', description: 'Busy workers threshold', proc: proc(&:to_i), default: 150, required: false
+  option :uptime, long: '--uptime UPTIME', short: '-u UPTIME', description: 'UPTIME threshold in seconds', proc: proc(&:to_i), default: 60, required: false
+  option :load, long: '--load LOAD', short: '-l LOAD', description: 'CPU Load threshold.', proc: proc(&:to_f), default: 80, required: false
+  option :debug, long: '--debug', short: '-d', description: 'Debug output', boolean: true, default: false, required: false
+  option :host, short: '-h HOST', long: '--host HOST', description: 'HOST to check mod_status output', default: '127.0.0.1'
+  option :port, short: '-p PORT', long: '--port PORT', description: 'Port to check mod_status output', default: '80'
+  option :secureport, long: '--secureport PORT', description: 'SSL Port to check mod_status output', default: '443'
+  option :path, short: '-path PATH', long: '--path PATH', description: 'PATH to check mod_status output', default: '/server-status?auto'
+  option :user, short: '-user USER', long: '--user USER', description: 'User if HTTP Basic is used'
+  option :password, short: '-password USER', long: '--password USER', description: 'Password if HTTP Basic is used'
+  option :secure, short: '-s', long: '--secure', description: 'Use SSL'
+
+  # Acquire Apache mod_status
+  def acquire_mod_status
+    http = Net::HTTP.new(config[:host], config[:port])
+    if config[:secure]
+      http = Net::HTTP.new(config[:host], config[:secureport])
+      http.verify_mode = OpenSSL::SSL::VERIFY_NONE
+      http.use_ssl = true
+    end
+    req = Net::HTTP::Get.new(config[:path])
+    if !config[:user].nil? && !config[:password].nil?
+      req.basic_auth config[:user], config[:password]
+    end
+    res = http.request(req)
+    case res.code
+    when '200'
+      res.body
+    else
```

Comment:
something like setting a max number of redirects and a case statement something like this:
```
when Net::HTTPRedirection then
  location = response['Location']
  # rerun the request here with the location
end
```